R/beers-r6.R

```diff
@@ -0,0 +1,153 @@
+
+  bf = matrix(nrow = 6, ncol = 6, byrow = TRUE, data = c(
+    0.0000,  0.0000,  0.0000,  0.0000,  1.0000,  0.0000,
+    0.0012, -0.0066,  0.0464, -0.1916,  1.0644,  0.0862,
+    0.0023, -0.0136,  0.0874, -0.3236,  1.0279,  0.2196,
+    0.0028, -0.0173,  0.1052, -0.3598,  0.8529,  0.4099,
+    0.0022, -0.0140,  0.0820, -0.2640,  0.5270,  0.6668,
+    0.0000,  0.0000,  0.0000,  0.0000,  0.0000,  1.0000))
+)
+
+beers_ordinary_r6 <- Beers_Interpolator$new(
+  b1 = matrix(nrow = 5, ncol = 6, byrow = TRUE, data = c(
+    1.0000,  0.0000,  0.0000,  0.0000,  0.0000,  0.0000,
+    0.6667,  0.4969, -0.1426, -0.1006,  0.1079, -0.0283,
+    0.4072,  0.8344, -0.2336, -0.0976,  0.1224, -0.0328,
+    0.2148,  1.0204, -0.2456, -0.0536,  0.0884, -0.0244,
+    0.0819,  1.0689, -0.1666, -0.0126,  0.0399, -0.0115)),
+
+  b2 = matrix(nrow = 5, ncol = 6, byrow = TRUE, data = c(
+     0.0000,  1.0000,  0.0000,  0.0000,  0.0000,  0.0000,
+    -0.0404,  0.8404,  0.2344, -0.0216, -0.0196,  0.0068,
+    -0.0497,  0.6229,  0.5014, -0.0646, -0.0181,  0.0081,
+    -0.0389,  0.3849,  0.7534, -0.1006, -0.0041,  0.0053,
+    -0.0191,  0.1659,  0.9354, -0.0906,  0.0069,  0.0015)),
+
+  bm = matrix(nrow = 5, ncol = 6, byrow = TRUE, data = c(
+    0.0000,  0.0000,  1.0000,  0.0000,  0.0000,  0.0000,
+    0.0117, -0.0921,  0.9234,  0.1854, -0.0311,  0.0027,
+    0.0137, -0.1101,  0.7194,  0.4454, -0.0771,  0.0087,
+    0.0087, -0.0771,  0.4454,  0.7194, -0.1101,  0.0137,
+    0.0027, -0.0311,  0.1854,  0.9234, -0.0921,  0.0117)),
+
+  bp = matrix(nrow = 5, ncol = 6, byrow = TRUE, data = c(
+    0.0000,  0.0000,  0.0000,  1.0000,  0.0000,  0.0000,
+    0.0015,  0.0069, -0.0906,  0.9354,  0.1659, -0.0191,
+    0.0053, -0.0041, -0.1006,  0.7534,  0.3849, -0.0389,
+    0.0081, -0.0181, -0.0646,  0.5014,  0.6229, -0.0497,
+    0.0068, -0.0196, -0.0216,  0.2344,  0.8404, -0.0404)),
+
+  bf = matrix(nrow = 6, ncol = 6, byrow = TRUE, data = c(
+    0.0000,  0.0000,  0.0000,  0.0000,  1.0000,  0.0000,
+   -0.0115,  0.0399, -0.0126, -0.1666,  1.0689,  0.0819,
+   -0.0244,  0.0884, -0.0536, -0.2456,  1.0204,  0.2148,
+   -0.0328,  0.1224, -0.0976, -0.2336,  0.8344,  0.4072,
+   -0.0283,  0.1079, -0.1006, -0.1426,  0.4969,  0.6667,
+    0.0000,  0.0000,  0.0000,  0.0000,  0.0000,  1.0000))
+)
+
+#' Beers modified algorithm for interpolation.
+#'
+#' @param points A list of at least 6 numbers.
+#' @return A list where each original number exists with four interpolated points between each.
+#' @export
+#' @examples
+#' beers_modified(c(1:6)) returns 1.0 1.2 1.4 1.6 1.8 2.0 2.2 ... 5.8 6.0
+
+beers_modified <- function(points) {
+  beers_modified_r6$interpolate(points)
+}
+
+#' Beers ordinary algorithm for interpolation.
+#'
+#' @param points A list of at least 6 numbers.
+#' @return A list where each original number exists with four interpolated points between each.
+#' @export
+#' @examples
+#' beers_ordinary(c(1:6)) returns 1.0 1.2 1.4 1.6 1.8 2.0 2.2 ... 5.8 6.0
```

Comment:
This example will not run!  So this will make your package fail the checks (try `devtools::check()` to see!)